script/bin/operator.rs

```diff
@@ -45,49 +44,353 @@ sol! {
+            let id_display_str = ids
+                .iter()
+                .map(|id| id.to_string())
+                .collect::<Vec<String>>()
+                .join(", ");
+
+            let this = self.clone();
+
+            // Spawn a task for each starting block, to compute the proofs concurrently.
+            let fut = async move {
+                tokio::spawn({
+                    async move {
+                        let this = this.clone();
+                        this.handle_batch(&ids, last_known_block, block_to_request)
+                            .await
+                    }
+                    .instrument(tracing::span!(
+                        tracing::Level::INFO,
+                        "compute_batch_proof",
+                        chains = id_display_str
+                    ))
+                })
+                .await
+                .expect("Join error")
+            };
+
+            handles.push(fut);
+        }
+
+        // Individually check each task for errors.
+        let results = futures::future::join_all(handles).await;
+
+        // Errors either occur when creating proofs or when relaying proofs.
+        //
+        // In either case, indicate that the operator should retry sooner.
+        let mut has_err = false;
+        for batch_result in results {
+            match batch_result {
+                Ok(relay_results) => {
+                    for relay_result in relay_results {
+                        if let Err(e) = relay_result {
+                            tracing::error!("Error relaying proof: {:?}", e);
+                            has_err = true;
+                        }
+                    }
+                }
+                Err(e) => {
+                    tracing::error!("Error running batch: {:?}", e);
+                    has_err = true;
+                }
+            }
+        }
+
+        if has_err {
+            // Any errors would have been logged already.
+            //
+            // Return an indicator to retry sooner.
+            return Err(anyhow::anyhow!(""));
+        }
+
+        Ok(())
+    }
+
+    /// Run the operator, indefinitely.
+    async fn run(self) {
+        let this = Arc::new(self);
+
+        tracing::info!("Operator running with chains {:?}", this.contracts.keys());
+
+        loop {
+            let request_interval_mins = get_loop_interval_mins();
+
+            tokio::select! {
+                _ = tokio::time::sleep(tokio::time::Duration::from_secs(60 * LOOP_TIMEOUT_MINS)) => {
+                    tracing::error!("Operator took longer than {} minutes to run.", LOOP_TIMEOUT_MINS);
+                    continue;
+                }
+                res = this.clone().run_inner().instrument(tracing::span!(tracing::Level::INFO, "operator")) => {
+                    if let Err(e) = res {
+                        tracing::error!("Error running operator: {:?}", e);
+
+                        // Sleep for less time on errors.
```

Comment:
```suggestion
                        // If there's an error, sleep for only 10 seconds. This will avoid transient RPC downtime while ensuring that the operator recovers quickly.
```